### ISSUE TITLE
Add extension directory for external tools

### ISSUE THREAD

**Issue by 0pendev** [flathub/org.gnu.emacs#79]
Using the flatpak for a few weeks now and the only drawback I see from something like the visual studio code extension is the lack for a way to easily package additional tools. The visual studio code flatpak adds this support with:

1. a directory extension making it possible for additional software to be run in the flatpak environment (useful to run containers with podman-remote for example)
2. a host-spawn command to easily run a shell on the host

I think the extension directory would be an easy way to add more usage of the package without having a great cost in support.

**Comment by slahtine** [comment]
You can implement one fairly easily yourself if you want. I have a `~/.var/app/org.gnu.emacs/data/local/bin` directory with some tools which are just `flatpak-spawn --host` scripts, but they could also be regular binaries. They should be exposed to Emacs under `/var/data/local/bin`, so you just need to add them to `PATH` and/or `exec-path`.

I have the following in my config to do it:

```elisp
(setenv "PATH" (concat (getenv "PATH") ":" (expand-file-name "/var/data/local/bin")))
(setq exec-path (append exec-path (list (expand-file-name "/var/data/local/bin"))))
```

**Comment by 0pendev** [comment]
Yes this is indeed possible but my point was to make the flatpak itself compatible with flatpak extensions. That could be interesting to have a standard wat to extend the emacs environment on flathub (just as VsCode does it)
